web/fido_app/templates/register.html

```diff
@@ -8,15 +8,19 @@
 
 {% block cardBody %}
 <!--Registration Form-->
-<form id="registration-form" class="primary-form" method="POST" timed-element timed-events="submit">
+<form id="registration-form" class="primary-form" method="POST" action="{% if current_user.is_authenticated %}/add-password{% else %}/register{% endif %}" timed-element timed-events="submit">
 
   <!-- CSRF Protection -->
   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
 
-  <div class="form-group">
-    <label for="email">Email address</label>
-    <input name="email" type="email" class="form-control" id="email" autocomplete="email" required timed-element />
-  </div>
+  {% if current_user.is_authenticated %}
```

Comment:
As mentioned in my other comments, my idea to use a hidden form field for the user's email is insecure (without additional verification). So we shouldn't need to use hidden form fields here.